drake/examples/Pendulum/Pendulum.h

```diff
@@ -3,100 +3,128 @@
 
 #include <iostream>
 #include <cmath>
-#include "DrakeSystem.h"
-#include "LCMCoordinateFrame.h"
-#include "BotVisualizer.h"
+#include "System.h"
+#include "LQR.h"
 
-class Pendulum : public DrakeSystem {
-public:
-  Pendulum(const std::shared_ptr<lcm::LCM>& lcm) :
-          DrakeSystem("Pendulum",
-                  std::make_shared<CoordinateFrame>("PendulumContState",std::vector<std::string>({"theta","thetadot"})),
-                  nullptr,
-                  std::make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumInput",std::vector<std::string>({"tau"}),lcm),
-                  std::make_shared<LCMCoordinateFrame<drake::lcmt_drake_signal> >("PendulumState",std::vector<std::string>({"theta","thetadot"}),lcm)),
-          m(1.0), // kg
-          l(.5),  // m
-          b(0.1), // kg m^2 /s
-          lc(.5), // m
-          I(.25), // m*l^2; % kg*m^2
-          g(9.81) // m/s^2
-  {}
-  virtual ~Pendulum(void) {};
+using namespace std;
 
-  virtual Eigen::VectorXd dynamics(double t, const Eigen::VectorXd& x, const Eigen::VectorXd& u) const override {
-    return dynamics_implementation(t,x,u);
-  }
-  virtual Drake::TaylorVecX dynamics(Drake::TaylorVarX t, const Drake::TaylorVecX& x, const Drake::TaylorVecX& u) const override {
-    return dynamics_implementation(t,x,u);
+template <typename ScalarType = double>
+class PendulumState  { // models the Drake::Vector concept
+public:
+  PendulumState(void) : theta(0), thetadot(0) {};
+  template <typename Derived>
+  PendulumState(const Eigen::MatrixBase<Derived>& x) : theta(x(0)), thetadot(x(1)) {};
+
+  template <typename Derived>
+  PendulumState& operator=(const Eigen::MatrixBase<Derived>& x) {
+    theta = x(0);
+    thetadot = x(1);
+    return *this;
   }
 
-  virtual Eigen::VectorXd output(double t, const Eigen::VectorXd& x, const Eigen::VectorXd& u) const override {
+  operator Eigen::Matrix<ScalarType,2,1> () const {
+    Eigen::Matrix<ScalarType,2,1> x;
+    x << theta, thetadot;
     return x;
   }
 
-  virtual bool isTimeInvariant() const override { return true; }
-  virtual bool isDirectFeedthrough() const override { return false; }
+  friend std::ostream& operator<<(std::ostream& os, const PendulumState& x)
+  {
+    os << "  theta = " << x.theta << endl;
+    os << "  thetadot = " << x.thetadot << endl;
+    return os;
+  }
 
+  enum {
+    RowsAtCompileTime = 2
+  };
+  static std::size_t size() { return 2; }
```

Comment:
I don't think we can always have this be static. For an RBM loaded from a URDF, we won't know the size until runtime and we need to pick a state vector type at compile time, so the size would probably be passed into the constructor of the state vector type, meaning it isn't necessarily the same for all instances of the class.

We can fix that later though.